maas/client/viscera/fabrics.py

```diff
@@ -0,0 +1,82 @@
+"""Objects for fabrics."""
+
+__all__ = [
+    "Fabrics",
+    "Fabric",
+]
+
+from . import (
+    check,
+    check_optional,
+    Object,
+    ObjectField,
+    ObjectSet,
+    ObjectType,
+)
+
+
+class FabricsType(ObjectType):
+    """Metaclass for `Fabrics`."""
+
+    async def read(cls):
+        data = await cls._handler.read()
+        return cls(map(cls._object, data))
+
+    async def create(cls, name: str, description: str, class_type: str):
+        """
+        Create a `Fabric` in MAAS.
+
+        :param name: The name of the `Fabric`.
+        :type name: `str`
+        :param description: A description of the `Fabric`.
+        :type description: `str`
+        :param class_type: The class type of the `Fabric`.
+        :type class_type: `str`
```

Comment:
name, description and class_type should be optional. You can create a fabric with no parameters.